src/components/IndexTable/IndexTable.tsx

```diff
@@ -118,21 +125,57 @@ function IndexTableBase({
     );
   }, [handleSelectionChange, selectedItemsCount]);
 
-  const resizeTableHeadings = useCallback(() => {
-    if (!tableElement.current || !scrollableContainerElement.current) {
-      return;
-    }
+  const calculateFirstHeaderOffset = useCallback(() => {
+    return condensed
+      ? tableHeadingRects.current[0].offsetWidth
+      : tableHeadingRects.current[0].offsetWidth +
+          tableHeadingRects.current[1].offsetWidth;
+  }, [condensed]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const resizeTableHeadings = useCallback(
+    throttle(() => {
+      if (!tableElement.current || !scrollableContainerElement.current) {
+        return;
+      }
 
-    const measuredTableHeadingRects = Array.from(
-      tableElement.current.querySelectorAll('[data-index-table-heading]'),
-    ).map((heading) => ({
-      offsetWidth: heading instanceof HTMLElement ? heading.offsetWidth : 0,
-      offsetLeft: heading instanceof HTMLElement ? heading.offsetLeft : 0,
-    }));
-    const boundingRect = scrollableContainerElement.current.getBoundingClientRect();
-    tablePosition.current = {top: boundingRect.top, left: boundingRect.left};
-    tableHeadingRects.current = measuredTableHeadingRects;
-  }, []);
+      const boundingRect = scrollableContainerElement.current.getBoundingClientRect();
+      tablePosition.current = {top: boundingRect.top, left: boundingRect.left};
+
+      tableHeadingRects.current = tableHeadings.current.map((heading) => ({
+        offsetWidth: heading.offsetWidth || 0,
+        offsetLeft: heading.offsetLeft || 0,
+      }));
+
+      if (tableHeadings.current.length === 0) {
+        return;
+      }
+
+      // update left offset for first column
+      if (tableHeadings.current.length > 1)
+        tableHeadings.current[1].style.left = `${tableHeadingRects.current[0].offsetWidth}px`;
+
+      // update the min width of the checkbox to be the be the un-padded width of the first heading
+      if (stickyHeaderCheckboxElement?.current) {
+        const elementStyle = getComputedStyle(tableHeadings.current[0]);
+        const boxWidth = tableHeadings.current[0].offsetWidth;
+        stickyHeaderCheckboxElement.current.style.minWidth = `calc(${boxWidth}px - ${elementStyle.paddingLeft} - ${elementStyle.paddingRight} + 2px)`;
+      }
+
+      // update sticky header min-widths
+      stickyTableHeadings.current.forEach((heading, index) => {
+        let minWidth = 0;
+        if (index === 0) {
+          minWidth = calculateFirstHeaderOffset();
+        } else if (tableHeadingRects.current.length > index) {
+          minWidth = tableHeadingRects.current[index].offsetWidth;
+        }
+
+        heading.style.minWidth = `${minWidth}px`;
```

Comment:
this could cause N layouts, where N = number of columns. if you can find a way to store list data in a css var, you could build up a set of css vars and then set them 1 time on a common parent, and avoid a bunch of layouts. not sure if it's possible though.